feat(media-library): add media library component

Davidmattei opened the pull request:
|Q              |A  |
|---------------|---|
|Bug fix?       |n|
|New feature?   |y|
|BC breaks?     |n|
|Deprecations?  |n|
|Fixed tickets  |n|

Implementation

```twig
{{ component('media_library', {
    'id': 'my-media-id',
   'contentTypeName': 'file',
   'field_path': 'media_path',
   'field_alpha_order': 'media_path.alpha_order',
   'field_location': 'media_folder',
   'field_file': 'media_file'
}) }}
````

Example created document:
```json
{
  "_source": {
    "_contenttype": "file",
    "_finalization_datetime": "2022-12-14T18:29:44+0100",
    "_finalized_by": "user",
    "media_file": {
      "filename": "file.png",
      "filesize": 438,
      "mimetype": "image/png",
      "sha1": "5aed565e04b5c16d11b7471b2e7c6cc004da3931"
    },
    "media_folder": "/images/sub/",
    "media_path": "/images/sub/file.png",
    "_sha1": "0a452e480d75e3c7d9068c9de4a222d7c32b0e45",
    "_published_datetime": "2022-12-14T18:29:44+0100"
  }
}
````

`field_location` is required for searching everything inside an exact path. Its the `field_path` without foldername or filename.

Davidmattei committed:
feat(media-library): add ux twig component

Davidmattei committed:
feat(media-library): implement core config with param converter

Davidmattei committed:
feat(media-library): config should be created by factories

Davidmattei committed:
feat(media-library): add abstract config factory

Davidmattei committed:
feat(media-library): config contains hash

It is not the responsibility of the component, but the config factories should create the hash

Davidmattei committed:
feat(media-library): added getFiles action in MediaLibrary controller

AbstractConfigFactory is smarter now, ConfigFactory should not handle the hash creation

Davidmattei committed:
feat(media-library): getFiles javascript

Davidmattei committed:
feat(media-library): add new translation domain emsco-component

Davidmattei committed:
feat(media-library): refactor core alerts.twig

Required for printing the alerts inside a modal after an ajax call

Davidmattei committed:
feat(media-library): add folder

Davidmattei committed:
feat(media-library): mediaLibrary.js options for ajaxUrlPath

The library should not know the path, this is an option on implementation

Davidmattei committed:
feat(media-library): clear flashBag

Example: adding 10 folders will give 10 info messages on page reload.
Page is not reloaded when the folder was added.

Davidmattei committed:
feat(media-library): getFolders and display them

Davidmattei committed:
feat(media-library): add "id" config for media library

Davidmattei committed:
feat(media-library): add progessBar helper

Davidmattei committed:
feat(media-library): upload file

Davidmattei committed:
feat(media-library): fix bug UploadedAssetRepository.php

theus77 reviewed (commented):
I did not understood (on my smartphone) how the js for JsonMenu* was loaded.

Davidmattei committed:
feat(media-library): print files with url and refactor styling

Davidmattei committed:
feat(media-library): recursive folders plus styling

Davidmattei committed:
feat(media-library): remove js timeout and extra styling

Davidmattei committed:
feat(media-library): refresh index after creation

Davidmattei committed:
feat(media-library): add breadcrumb and open folder after creation

Davidmattei committed:
feat(media-library): reword folder to path and on doc location

Davidmattei committed:
feat(media-library): add alpha order field

If not defined, no sorting

Davidmattei committed:
feat(media-library): clean js & css

On ajax call disable all buttons, everything is a button inside the media library.

Davidmattei committed:
build: assets

Davidmattei committed:
Merge branch '4.x' into feat/media-library

theus77 commented on EMS/core-bundle/src/Core/Component/MediaLibrary/MediaLibraryConfigFactory.php:
```diff
@@ -0,0 +1,73 @@
+    {
+        $resolver = new OptionsResolver();
+        $resolver
+            ->setDefaults([
```
No setType?

Davidmattei committed:
Merge branch '4.x' into feat/media-library
